Dockerfile

```diff
@@ -1,18 +1,35 @@
-FROM quay.io/app-sre/fluentd-upstream:v1.16-1
+FROM registry.redhat.io/openshift-logging/fluentd-rhel9@sha256:a31038fa2b953660e09b7db8411e808075c9037b679ba56a4a6245f56853db56
 
 USER root
 
-RUN apk update && apk upgrade busybox busybox-binsh libcrypto3 libssl3 \
-    ncurses-libs ncurses-terminfo-base ssl_client \
-    && apk add --no-cache --update --virtual .build-deps build-base ruby-dev \
-    && echo 'gem: --no-document' >> /etc/gemrc \
-    && gem install fluent-plugin-s3 \
-    && gem install fluent-plugin-slack \
-    && gem install fluent-plugin-cloudwatch-logs \
-    && gem install fluent-plugin-teams \
-    && gem install fluent-plugin-rewrite-tag-filter \
-    && gem install nokogiri \
-    && apk del .build-deps \
-    && rm -rf /tmp/* /var/tmp/* /usr/lib/ruby/gems/*/cache/*.gem
+RUN microdnf install -y \
+    gcc \
+    make \
+    ncurses-base \
+    ncurses-libs \
+    redhat-rpm-config \
+    ruby-devel \
+    openssl \
+    openssl-libs \
+ && microdnf clean all
 
+# skip installing documentation
+RUN echo 'gem: --no-document' >> /etc/gemrc
+
+RUN curl -L https://busybox.net/downloads/binaries/1.36.0-defconfig-multiarch/busybox-x86_64 \
+    -o /usr/local/bin/busybox && \
+    chmod +x /usr/local/bin/busybox && \
+    ln -s /usr/local/bin/busybox /usr/local/bin/sh
```

Comment:
why we need busybox?

Reply:
Not 100% if I'm honest, the current Dockerfile has it installed so I was keeping the same pattern but happy to remove it completely if it's something we don't use

Reply:
After looking at [this](https://github.com/app-sre/fluentd/commit/387104acfcfe1f4fd81cfb7b5c85dcb974bf01a9) and talking to Jon, the original upstream fluentd image included buxybox, and the current app-sre dockerfile is update it to resolve CVEs.

If busybox isn't on the new fluentd-rhel9 image, I think we can remove the busybox download step here.